app/javascript/components/patient/assessment/actions/SymptomOnset.js

```diff
@@ -61,53 +56,111 @@ class SymptomOnset extends React.Component {
     });
   };
 
+  closeModal = () => {
+    this.setState({
+      symptom_onset: this.props.patient.symptom_onset,
+      user_defined_symptom_onset: this.props.patient.user_defined_symptom_onset,
+      showSymptomOnsetModal: false,
+      first_positive_lab: null,
+      showLabModal: false,
+    });
+  };
+
+  createModal = (message, invalid, close, submit) => {
+    return (
+      <Modal size="lg" show centered onHide={close}>
+        <Modal.Header>
+          <Modal.Title>Symptom Onset</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <p>{message}</p>
+          {invalid && (
+            <React.Fragment>
+              <p className="mt-2 mb-3">
+                Since the Symptom Onset Date will be set to blank,{' '}
+                {this.state.first_positive_lab ? 'the following lab result will be created' : 'please enter a positive lab result'} in order for this record to
+                be eligible to appear on the Records Requiring Review line list as an asymptomatic case.
+              </p>
+              <FirstPositiveLab lab={this.state.first_positive_lab} onChange={lab => this.setState({ first_positive_lab: lab })} />
+            </React.Fragment>
+          )}
+        </Modal.Body>
+        <Modal.Footer>
+          <Button variant="secondary btn-square" onClick={close}>
+            Cancel
+          </Button>
+          <Button variant="primary btn-square" onClick={submit} disabled={this.state.loading}>
+            {this.state.loading && (
+              <React.Fragment>
+                <span className="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>&nbsp;
+              </React.Fragment>
+            )}
+            <span data-for="lde-submit" data-tip="">
+              Submit
+            </span>
+          </Button>
+        </Modal.Footer>
+      </Modal>
+    );
+  };
+
   render() {
+    const symptom_onset_invalid =
+      this.props.patient.isolation && !this.state.symptom_onset && !this.props.symptomatic_assessments_exist && this.props.num_pos_labs === 0;
     return (
       <React.Fragment>
-        <Form.Group as={Col} controlId="symptom_onset">
-          <Row className="reports-actions-title">
-            <Col>
-              <Form.Label className="nav-input-label">
-                SYMPTOM ONSET
-                <InfoTooltip tooltipTextKey={this.props.patient.isolation ? 'isolationSymptomOnset' : 'exposureSymptomOnset'} location="right"></InfoTooltip>
-                <div style={{ display: 'inline' }}>
-                  <span data-for="user_defined_symptom_onset_tooltip" data-tip="" className="ml-2">
-                    {this.props.patient.user_defined_symptom_onset ? <i className="fas fa-user"></i> : <i className="fas fa-desktop"></i>}
+        {this.state.showSymptomOnsetModal &&
```

Comment:
can we break this logic out into a createModal method? would look a lot cleaner and I assume there's no reason it can't be added before the return

Reply:
https://github.com/SaraAlert/SaraAlert/pull/861/commits/0b00f9efd83684aa21526ddee93b433c20c8fb79#diff-c70b6a432860cf1cff1aa3bc624a2b72da37531257c62855dba8ee265610f6e9L112